### DISCUSSION TITLE
Move changes to a remote revision into a new revision

### DISCUSSION THREAD

**Discussion by jbcdnr** [jj-vcs/jj#2400] [Q&A] [answered]
Hello,

I sometimes get in trouble by editing a remote commit and I would like to move these changes into a new commit.

Example:
```bash
jj edit main # Should have done `jj co main` to create a new revision
# edit some files and save
jj log
# Now there is a * next to the branch main
# I can see the diff with my local changes
jj diff --from main@origin --to main
```
How can I restore the revision to main and put the changes in a new revision?

Thank you!

**Comment by martinvonz** [comment, marked as answer]
Since jj 0.10.0, it will try to prevent you from rewriting remote commits. You can [configure the set of commits you consider immutable](https://martinvonz.github.io/jj/v0.10.0/config/#set-of-immutable-commits).

There's also [an FAQ entry](https://martinvonz.github.io/jj/v0.10.0/FAQ/#i-accidentally-amended-the-working-copy-how-do-i-move-the-new-changes-into-its-own-commit) explaining how to recover if you've accidentally modified a non-remote commit.

But for your case, where you modified a remote commit, one way of fixing it is this:

1. Run `jj co main@origin` to create a new change on top of the remote branch. You will now have two commits with the same change id (aka divergence)
2. Run `jj restore --from main` to copy the state from that commit into your new change
3. Run `jj abandon main` since you don't want that version. This will resolve the divergence and move the branch backwards
4. Run `jj branch set main -r main@origin` (or `-r @-`) to move the branch back to where it was supposed to be

HTH

**Reply by necauqua** [reply]
> You will now have two commits with the same change id (aka divergence)

Will you?. Isn't it just a new commit, I think there wont be a divergent changeid here

~~Also a shorter solution could be to rebase main on top of main@origin and then do p. 4~~
To _reparent_ once we have a reparent function or similar)

**Reply by martinvonz** [reply]
I think so, because creating a new commit on top of the hidden `main@origin` will make it visible again. The commit `main` points to at this point has the same change id.

**Reply by jbcdnr** [reply]
This is really helpful. Thanks a lot, and thank you for the great tool! Improved our workflows so much :)

**Reply by necauqua** [reply]
> will make it visible again

ah, got it, that's absolutely correct, got confused by the wording I guess
